migrations/v0.7.0/README.md

```diff
@@ -0,0 +1,26 @@
+# v0.7.0 migration
+
+When migrating from Vela version [v0.6.0](https://github.com/go-vela/community/blob/master/releases/v0.6.0.md) to [v0.7.0](https://github.com/go-vela/community/blob/master/releases/v0.7.0.md)  there is a breaking change for all native secrets. The change was to [encrypt all secret values](https://github.com/go-vela/community/issues/100) stored in the native engine. This means the endpoints in v0.7.0 where not compatible to read/write old secrets with the new encryption process. Secrets that reside in the database need to be encrypted.
+
+Instead of building backwards compatibility we want all secrets to be encrypted which meant we need to supply a script that can be used during the version upgrade to assist admins with keeping their secrets create pre [v0.6.0](https://github.com/go-vela/community/blob/master/releases/v0.6.0.md) compatible with later versions. 
+
+**Script Requirements:**
+
+* [Go Installed](https://golang.org/doc/install)
+  * Along with packages referenced in `main.go` file in this directory
+  * Tip: packages can be downloaded with `go.mod`
+* Address to vela server
+* encryption key used for v0.7.0 installation
+* Vela user account token with admin access
+
+**Usage:**
+
+```sh
+# run the script within this directory
+export VELA_ADDR=<server_addr>
+export VELA_KEY=<server_key>
+export VELA_TOKEN=<admin_token>
+
+# execute program
+go run main.go
```

Comment:
Do we think it's worth providing the binary directly in this directory?

Not sure if it's a good idea to be embedding binaries into version control but it might help improve user experience?

Reply:
I'm not opposed to adding the binary, it's easy enough. My only concern would be to we have to supply it for all OS types (OSX, Linux, Windows, etc) then I wonder if that might be overkill. 


I think... 🤔 that they would have to clone the repo, either way, to execute the migration script/binary because I don't believe GitHub allows you to download a single file (binary in this case) from a repo.

Reply:
Sure that's fair 👍 

I'm not entirely sure embedding binaries into the repo isa good idea anyways.

Maybe for now we leave it out but if someone else requests it then we'll add it?

Reply:
How about creating some executable docker image that contains the binary? e.g.

```dockerfile
FROM alpine

ADD release/convert /bin/convert
```

```sh
docker exec -it target/vela-convert:0.7.0 sh
/bin/convert <option1> <option2> ...
```

Reply:
@JordanSussman I like that idea 👍

Reply:
I can do that 😄 

Does it make sense to pull these files into a new repo like `go-vela/migrations` that way this stuff can be in automation?

Or is it fair to just build that image once locally and push it?

Reply:
@kneal good question 😅 

Hmm... I like the idea of a `migrations` repo 👍 

Although I think you could still do the automation in this repo with an Actions pipeline that builds a Docker image.

You could have the `Dockerfile` for that image located right in the `v0.7.0` directory alongside the `main.go` and `README.md` etc.

I'm not sure I have a preference as I see value from both sides.

I think it would be nice to have 1 spot for all the files so having it in this repo would be nice..

But if we use this as a pattern going forward, I could see this repo getting larger than we initially intended as we add more `migrations` for different versions?

Thoughts?

Reply:
I would prefer to utilize this repo to reduce the already confusing number of places to look for relevant code.

Reply:
I think it's worth calling out where ever we do this, it's going to look kinda mono repo ish which will be kinda new for our automation. So, when I push up the files we can chat more about design.

Reply:
@JordanSussman @jbrockopp 

So, circling back to this to wrap this up. Outstanding items are

- [ ] Add a Dockerfile
- [ ] Update docs for docker image usage
- [ ] _Add CI for Docker image?_

To me the CI aspect of this doesn't seem that valuable? Once we get the script in the image and published I can't imagine this seems like something we'll care about continually updating long term? Would you be open to a one time publish? I personally feel while there is value in making this as user friendly as possible. If an admin can't build an image with `docker build -t` etc, they will probably have a tough time maintaining the infrastructure to deploy Vela. 

If it is something we want in CI, do you have any thoughts on how the publish would work? We don't really have an established workflow for this kinda of repo (tag, push to master, etc). I guess my assumption would be I would just do a push on merge to master and always overwrite a `target/vela-migrations/encryption:v0.7.0` tag?

Reply:
I'm personally fine with having the administrator run commands to generate the docker image. e.g.

```sh
git clone https://github.com/go-vela/community
cd community
git checkout v0.7.x
docker build -t vela-migration:latest
docker run --rm -it vela-migration:latest sh
...
```

Reply:
I think a `Makefile` is 💯 doable. That way we are setup for CI if need be. I've updated my list above with the feedback. Thanks!

Reply:
Let me know what y'all think of this change:
https://github.com/go-vela/community/pull/154/commits/2182734f42b6cd4b0fdab2ae5158991f9cf3d674

Reply:
Probably best to add https://github.com/go-vela/community/blob/2182734f42b6cd4b0fdab2ae5158991f9cf3d674/migrations/v0.7.0/release/vela-migration to the `.gitignore` file.